### PR TITLE
Facilitate user experience

### PR DESCRIPTION
**Pull request by wyhasany**
With newest versions of intellij the pop up informing that annotation
processor is disabled won't work. Hence I've fixed checking if
proper checkbox has been marked. Moreover current use experience was
bad as user had to expand notification memoize instruction then
config window has appeared (while instruction disappears) so user has
been in stuck. To resolve that I'm showing more laconic notification
which enables Annotation Processor for user automatically.

tldr;
Before:

Selection_117

then

Selection_115

then no one knows what to do here:

Selection_118

After:
Selection_116

then nice balloon informs us that everything has been made for us:

Selection_119

### CODE REVIEW

**Comment by mplushnikov** [comment]
Hello @wyhasany 
thank you for your contribution!

Please reformat your commit usind project specific codestyle. You can use EditorConfig file to import it from project. (http://EditorConfig.org)

Also please add yourself to the list of contributors in Readme.md file.

**Comment by wyhasany** [comment]
Hmm, that's so strange as I applied your `.editorconfig` codestyle. I don't see any violation of it. Could you point me out what is wrong with code style? I've updated README as you asked.

File: src/main/java/de/plushnikov/intellij/plugin/activity/LombokProjectValidatorActivity.java [hunk 1]
```diff
@@ -79,29 +81,47 @@ public void runActivity(@NotNull Project project) {
...
+      suggestEnableAnnotations(project, group);
+    }
+  }
 
-      Notification notification = group.createNotification(LombokBundle.message("config.warn.annotation-processing.disabled.title"),
-        LombokBundle.message("config.warn.annotation-processing.disabled.message", project.getName()),
-        NotificationType.ERROR,
-        new SettingsOpeningListener(project, annotationProcessorsConfigName));
+  private void suggestEnableAnnotations(Project project, NotificationGroup group) {
+    String annotationProcessorsConfigName = new AnnotationProcessorsConfigurable(project).getDisplayName();
```
Comment (mplushnikov) [inline src/main/java/de/plushnikov/intellij/plugin/activity/LombokProjectValidatorActivity.java]: This variable looks like being obsolete, it is not used anywhere?

File: src/main/java/de/plushnikov/intellij/plugin/activity/LombokProjectValidatorActivity.java [hunk 2]
```diff
@@ -79,29 +81,47 @@ public void runActivity(@NotNull Project project) {
...
 
-      Notifications.Bus.notify(notification, project);
-    }
+    Notification notification = group.createNotification(LombokBundle.message("config.warn.annotation-processing.disabled.title"),
+      LombokBundle.message("config.warn.annotation-processing.disabled.message", project.getName()),
+      NotificationType.ERROR,
+      (not, e) -> {
+        if (e.getEventType() == HyperlinkEvent.EventType.ACTIVATED) {
+          enableAnnotations(project);
```
Comment (mplushnikov) [inline src/main/java/de/plushnikov/intellij/plugin/activity/LombokProjectValidatorActivity.java]: I think we can/should make additionally something like `not.expire();` here

File: src/main/java/de/plushnikov/intellij/plugin/activity/LombokProjectValidatorActivity.java (hunk 1 above)
Comment (wyhasany) [reply src/main/java/de/plushnikov/intellij/plugin/activity/LombokProjectValidatorActivity.java]: Yeah I've forgotten about it. I'll remove it :)

File: src/main/java/de/plushnikov/intellij/plugin/activity/LombokProjectValidatorActivity.java (hunk 2 above)
Comment (wyhasany) [reply src/main/java/de/plushnikov/intellij/plugin/activity/LombokProjectValidatorActivity.java]: is that necessary? Notification exprires even without calling this method.

Comment (mplushnikov) [reply src/main/java/de/plushnikov/intellij/plugin/activity/LombokProjectValidatorActivity.java]: I just tested it, and I think if looks as good additional feedback, if notification expires after click on "Enable"-Link.
Without expire you can click multiple times on it.

Comment (wyhasany) [reply src/main/java/de/plushnikov/intellij/plugin/activity/LombokProjectValidatorActivity.java]: great :+1:  I understand it now

File: src/main/java/de/plushnikov/intellij/plugin/activity/LombokProjectValidatorActivity.java (hunk 1 above)
Comment (wyhasany) [reply src/main/java/de/plushnikov/intellij/plugin/activity/LombokProjectValidatorActivity.java]: fixed

File: src/main/java/de/plushnikov/intellij/plugin/activity/LombokProjectValidatorActivity.java (hunk 2 above)
Comment (wyhasany) [reply src/main/java/de/plushnikov/intellij/plugin/activity/LombokProjectValidatorActivity.java]: fixed

**Comment by wyhasany** [comment]
It seems to be ready :)

**Comment by mplushnikov** [comment]
Thank you!

**Comment by wyhasany** [comment]
Thank you too for cooperation :)
